src/synapse-plugins/converter-plugin/converter-plugin-backend.vala

```diff
@@ -76,7 +76,8 @@ namespace Synapse {
 
             try {
                 convert_regex = new Regex (
-                    """(\d*) ?([a-zA-Z\/ ]+?)([23]?) ?[=\-]> ?([a-zA-Z\/ ]+)([23]?)""",
+                    //  Number? - space? - unit1 - dimension1 - space? => - space? - unit2 - dimension
+                    """^([[:digit:]]*\.?[[:digit:]]*)\s*([[:alpha:]\/ ]+)([23]?)\s?[=\-]>\s*([[:alpha:]\/ ]+?)([23]?)$""",
```

Comment:
1.  I made it greedy so that it would correctly match units with spaces in them e.g. `3US liquid pints => ml`
2. I didnt mean to make unit2 lazy - it needs changing
3. I thought `\d` only matched ASCII digits whereas `[:digit:]` matched localized digits as well? It is also more explicit.
4. Quite right - will fix.

Reply:
Hmm, tests still pass even with the lazy match and no stripping so I must have misunderstood what the regex is doing :disappointed:

Reply:
I have incorporated your suggestions, except re `[:digit:]`

Reply:
> Hmm, tests still pass even with the lazy match and no stripping so I must have misunderstood what the regex is doing disappointed

Yeah this confusing, it will lazy match until a space with arrow. But I should have included an example with a unit with spaces. 

All I can find is that `\d` and `[:digit:]` do the same. It's [0-9] that's more limited. But, if you think it's clearer let's keep it. 
https://www.regular-expressions.info/posixbrackets.html#class
https://unix.stackexchange.com/questions/414226/difference-between-0-9-digit-and-d